interfaces/console/src/app/console/sites/[id]/page.tsx

```diff
@@ -598,86 +462,196 @@ const Page: React.FC<SiteDetailsProps> = ({ params }) => {
       const to = getUnixTime();
       const from = to - STAT_STEP_29;
+      const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${from}`;
+
+      Object.keys(subscriptionsRef.current).forEach((topic) => {
+        if (
+          topic.startsWith(`stat-${user.orgName}-${user.id}-${Stats_Type.Site}`)
+        ) {
+          PubSub.unsubscribe(topic);
+          delete subscriptionsRef.current[topic];
+        }
+      });
+
+      subscriptionsRef.current[sKey] = true;
+
       getSiteMetricStat({
         variables: {
           data: {
-            to: to,
-            siteId: id,
-            from: from,
+            to,
+            from,
             userId: user.id,
             step: STAT_STEP_29,
             orgName: user.orgName,
             withSubscription: true,
             type: Stats_Type.Site,
+            siteIds: [id],
+            nodeIds: nodeIds,
+          },
+        },
+      });
+
+      MetricStatBySiteSubscription({
+        key: sKey,
+        siteIds: [id],
+        userId: user.id,
+        url: env.METRIC_URL,
+        orgName: user.orgName,
+        type: Stats_Type.Site,
+        from,
+        nodeIds: nodeIds,
+      });
+
+      return () => {
+        PubSub.unsubscribe(sKey);
+        delete subscriptionsRef.current[sKey];
+      };
+    }
+  }, [
+    id,
+    nodeIds,
+    nodesFetched,
+    user.id,
+    user.orgName,
+    env.METRIC_URL,
+    getSiteMetricStat,
+  ]);
+
+  useEffect(() => {
+    if (metricFrom > 0 && id) {
+      getMetricBySite({
+        variables: {
+          data: {
+            step: 30,
+            siteId: id,
+            userId: user.id,
+            type: graphType,
+            from: metricFrom,
+            orgName: user.orgName,
+            withSubscription: true,
+            to: metricFrom + METRIC_RANGE_10800,
           },
         },
       });
     }
+  }, [metricFrom, getMetricBySite, graphType, id, user.id, user.orgName]);
+
+  const handleSiteStatSubscription = useCallback(
+    (_: any, data: string) => {
+      try {
+        const parsedData: TMetricResDto = JSON.parse(data);
+        if (parsedData?.data?.getSiteMetricStatSub) {
+          const metric = parsedData.data.getSiteMetricStatSub;
+          const { type, success, siteId, nodeId, value } = metric;
+
+          if (success && siteId === id && !nodeId) {
+            if (type === SITE_KPI_TYPES.SITE_UPTIME) {
+              setSiteUptime(Math.floor(value[1]));
+            } else if (type === SITE_KPI_TYPES.SITE_UPTIME_PERCENTAGE) {
+              setSiteUptimePercentage(Math.floor(value[1]));
+            }
+          }
+
+          if (success && nodeId && type === 'unit_uptime') {
+            setNodeUptimes((prev) => {
+              return {
+                ...prev,
+                [nodeId]: Math.floor(value[1] || 1),
+              };
+            });
+          }
+
+          if (type && type !== 'unit_uptime') {
+            PubSub.publish(`stat-${type}`, value);
```

Comment:
Looks like we're doing a same thing here again. Here we're publishing to some topic are we subscribing to this topic inside components?

Reply:
addressed.